tck/src/main/java/ee/jakarta/tck/data/framework/utilities/TestProperty.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.data.framework.utilities;
+
+import java.util.logging.Logger;
+
+/**
+ * This class represents the different system properties used within this TCK.
+ * Each one is given a description and documentation will automatically be created in the TCK distribution.
+ */
+public enum TestProperty {    
+    //Java properties that should always be set by the JVM
+    javaHome    (true,  "java.home",                  "Path to the java executable used to create the current JVM"),
+    javaSpecVer (true,  "java.specification.version", "Specification version of the java executable"),
+    javaTempDir (true,  "java.io.tmpdir",             "The path to a temporary directory where a copy of the signature file will be created"),
+    javaVer     (true,  "java.version",               "Full version of the java executable"),
+    
+    //TCK specific properties
+    platform    (false, "jakarta.tck.platform",        "The platform name which can be appended to the test name for easier reporting"),
+    standalone  (true,  "jakarta.tck.standalone.test", "Set to true if running in standalone mode, false otherwise"),
```

Comment:
If you combine the above two properties into a profile that can either be web | core | none, with none meaning standalone, that would make it impossible to select invalid combinations like standalone & web profile toegether.

Reply:
Done 👍🏼